### DISCUSSION TITLE
How to generate xml file with version 5?

### DISCUSSION THREAD

**Discussion by sailboy1025** [jhu-dvrk/sawIntuitiveResearchKit#180] [Q&A] [answered]
I was calibrating the dVRK with sawRobotIO1394CurrentCalibration -c <xml.file>, however it asks for latest XML files. I couldn't find a config generator which gives me higher than version 4.

![Screenshot from 2023-06-11 11-01-01](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/124316691/e78228af-1d10-4c87-a50e-54ff9a4ab981)

The package is built with ros2 humble under 22.04 Ubuntu.

**Comment by adeguet1** [comment]
We added a python based config generator.  Unless I did something wrong, it should be in your path: dvrk_config_generator.py.  For the generation use Classic.  For the hardware type, use QLA1.  Note that I updated this code a few days ago so make sure use pull the latest code.   This code has not been thoroughly tested so I hope it'll work.

**Reply by sailboy1025** [reply]
Thanks for your responsive! So for MTMs, it prompts error like 'Unrecognized robot type: 64307 (which is our MTML serial number)', for PSMs and ECM it works fine, though all XML files are successfully generated.

For [general calibration](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/wiki/Calibration), is the `sawRobotIO1394CurrentCalibration` also deprecated? Since it works well on our old computer running with ros1 and version.4 XML, but wouldn't work with the XML I just generated for the reason ''E- File: osaXML1394.cpp Line: 407 -  All offsets equal, it is very unlikely that the current calibration has been performed for MTML", and this happened to all arms including ECM and PSMs.
![Screenshot from 2023-06-12 18-04-34](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/124316691/53de3c1a-e176-4ffb-a08a-cfe8c8e76005)

**Reply by adeguet1** [reply, marked as answer]
When did you pull the code?   I pushed some fixes for issues similar to the ones you mentioned last Friday.  Maybe do a vcs pull and rebuild?

**Reply by sailboy1025** [reply]
Yeah you're right, thank you sir!

**Reply by adeguet1** [reply]
To note, the Python script to perform the potentiometer calibration has not yet been ported to ROS 2.  I hope to do this early next week.  Also missing is the MTM gravity compensation code.  This might take a bit more time since it's Matlab based.
